File: app/src/main/java/com/divercity/app/features/profile/selectcompany/SelectCompanyActivity.java
package com.divercity.app.features.profile.selectcompany;

import android.content.Context;
import android.content.Intent;

import com.divercity.app.core.base.BaseFragment;
import com.divercity.app.core.base.BaseFragmentActivity;

/**
 * Created by lucas on 17/10/2018.
 */

public class SelectCompanyActivity extends BaseFragmentActivity {

    public static Intent getCallingIntent(Context context){
        return new Intent(context, SelectCompanyActivity.class);
    }

    @Override
    protected BaseFragment fragment() {
        return SelectCompanyFragment.newInstance();
    }
}

File: app/src/main/java/com/divercity/app/features/onboarding/selectcompany/SelectCompanyActivity.kt
package com.divercity.app.features.onboarding.selectcompany

import android.content.Context
import android.content.Intent
import com.divercity.app.core.base.BaseActivity
import com.divercity.app.core.base.BaseFragment

/**
 * Created by lucas on 17/10/2018.
 */

class SelectCompanyActivity : BaseActivity() {

    companion object {
        private const val INTENT_EXTRA_PARAM_PROGRESS = "progress"

        fun getCallingIntent(context: Context, progress : Int) : Intent {
            val intent = Intent(context, SelectCompanyActivity::class.java)
            intent.putExtra(INTENT_EXTRA_PARAM_PROGRESS,progress)
            return intent
        }
    }

    override fun fragment(): BaseFragment = SelectCompanyFragment.newInstance(
            intent.getIntExtra(INTENT_EXTRA_PARAM_PROGRESS, 0))
}
